VAULT-43958: Adding new fields to Vault Secrets Sync destination APIs (#2700)
hashicorp/web-unified-docs: content/vault/v2.x/content/api-docs/system/secrets-sync.mdx

- `region` `(string: "")` - Region where to manage the secrets manager entries. If omitted, configuration fallbacks on
the AWS credentials provider chain and tries to infer region from the environment.

> - `kms_key_id` `(string: "")` - The KMS key arn to be used for encrypting the secrets at rest in AWS Secrets Manager. 

Reviewer: ```suggestion
- `kms_key_id` `(string: "")` - The KMS key ARN Vault should use to encrypt secrets at rest in AWS Secrets Manager. 
```
Style correction: write in active voice

ARN is a proprietary acronym, so we should capitalize it per Amazon's usage

- `locational_kms_keys` `(map<string|string>: nil)` - A map of location names to KMS key names to leverage customer-managed encryption keys for
  encryption at rest. Each pair follows the format `location_name=encryption_key_resource_name`. Refer to the

>   [sample payloads](#sample-payloads) for more details. This field is deprecated. Use `regional_kms_keys` instead.

Reviewer: The deprecation notice should be at the start of the field, not the end. I can't add a suggestion since the first line wasn't included in the original edit, but I suggest rewriting as follows:

```
- `locational_kms_keys` `(map<string|string>: nil)` - **DEPRECATED** Use `regional_kms_keys` instead.
  A map of location names to KMS key names to leverage customer-managed encryption keys for
  encryption at rest. Each pair follows the format `location_name=encryption_key_resource_name`.
  Refer to the [sample payloads](#sample-payloads) for more details.
```

Author: resolved.

- `global_kms_key` `(string: "")` - The KMS key id to be used for encrypting the secrets at rest in GCP Secrets Manager. This field is deprecated. 

>    Use `kms_key_id` instead.

Reviewer: ```suggestion
- `global_kms_key` `(string: "")` - **DEPRECATED** Use `kms_key_id` instead.
   The KMS key ID Vault should use to encrypt secrets at rest in GCP Secrets Manager.
```

> - `kms_key_id` `(string: "")` - The KMS key id to be used for encrypting the secrets at rest in GCP Secrets Manager. 

Reviewer: ```suggestion
- `kms_key_id` `(string: "")` - The KMS key ID Vault should use for encrypting secrets at rest in GCP Secrets Manager. 
```
